Vagrantfile

```diff
@@ -112,19 +113,36 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
   dev_cluster = vagrant_openshift_config['dev_cluster'] || ENV['OPENSHIFT_DEV_CLUSTER']
   single_vm_cluster = ! (dind_dev_cluster or dev_cluster)
   if dind_dev_cluster
+
+    # /vagrant is used instead of /data because the vm is only the
+    # host for the cluster.  Members of a dind cluster will still use
+    # /data.
+    deployed_root="/vagrant"
+
     config.vm.define "#{VM_NAME_PREFIX}dind-host" do |config|
       config.vm.box = kube_box[kube_os]["name"]
       config.vm.box_url = kube_box[kube_os]["box_url"]
-      config.vm.provision "shell", inline: "/vagrant/contrib/vagrant/provision-dind.sh"
-      config.vm.provision "shell", inline: "/vagrant/hack/dind-cluster.sh config-host"
-      config.vm.provision "shell", privileged: false, inline: "/vagrant/hack/dind-cluster.sh restart"
+      config.vm.synced_folder ".", "/vagrant", disabled: true
+      config.vm.synced_folder ".", deployed_root, type: vagrant_openshift_config['sync_folders_type']
+      config.vm.provision "shell", inline: "#{deployed_root}/contrib/vagrant/provision-dind.sh"
+      config.vm.provision "shell", inline: "#{deployed_root}/hack/dind-cluster.sh config-host"
+      config.vm.provision "shell", privileged: false, inline: "#{deployed_root}/hack/dind-cluster.sh restart"
       config.vm.hostname = "openshift-dind-host"
-      config.vm.synced_folder ".", "/vagrant", type: vagrant_openshift_config['sync_folders_type']
     end
   elsif dev_cluster
     # Start an OpenShift cluster
     # Currently this only works with the (default) VirtualBox provider.
 
+    # Remove stale configuration before provisioning
+    if ARGV[0] =~ /^up|provision$/i and not ARGV.include?("--no-provision")
+      system('rm -rf ./openshift.local.*')
+    end
```

Comment:
It's only for a multinode dev cluster.  I thought I'd read somewhere along the way that 'halt' and then 'up' for a dev cluster doesn't work.  If that's not the case, I'm happy to remove this.